src/components/Auth/SignUp/SignUp.react.js

```diff
@@ -1,49 +1,85 @@
+// Packages
 import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+
+// Components
+import Recaptcha from 'react-recaptcha';
 import Paper from 'material-ui/Paper';
 import TextField from 'material-ui/TextField';
 import RaisedButton from 'material-ui/RaisedButton';
-import $ from 'jquery';
-import './SignUp.css';
 import PasswordField from 'material-ui-password-field';
 import Dialog from 'material-ui/Dialog';
-import PropTypes from 'prop-types';
-import UserPreferencesStore from '../../../stores/UserPreferencesStore';
 import zxcvbn from 'zxcvbn';
+import './SignUp.css';
+import UserPreferencesStore from '../../../stores/UserPreferencesStore';
 import Translate from '../../Translate/Translate.react';
-import Recaptcha from 'react-recaptcha';
 import { CAPTCHA_KEY } from '../../../config.js';
+import actions from '../../../redux/actions/app';
+
+const styles = {
+  paperStyle: {
+    width: '100%',
+    textAlign: 'center',
+    padding: '10px',
+  },
+  fieldStyle: {
+    height: '35px',
+    borderRadius: 4,
+    border: '1px solid #ced4da',
+    fontSize: 16,
+    padding: '0px 10px',
+    width: '250px',
+    marginTop: '10px',
+  },
+  inputStyle: {
+    height: '35px',
+    marginBottom: '10px',
+  },
+  inputpassStyle: {
+    height: '35px',
+    marginBottom: '10px',
+    marginRight: '50px',
+    width: '90%',
+  },
+};
+
+class SignUp extends Component {
+  static propTypes = {
+    history: PropTypes.object,
+    onRequestClose: PropTypes.func,
+    onLoginSignUp: PropTypes.func,
+    actions: PropTypes.object,
+  };
 
-export default class SignUp extends Component {
   constructor(props) {
     super(props);
 
     this.state = {
       email: '',
-      isEmail: false,
-      emailError: true,
-      passwordError: true,
-      passwordConfirmError: true,
-      passwordValue: '',
-      confirmPasswordValue: '',
-      msg: '',
-      success: false,
-      open: false,
-      openLogin: false,
-      openForgotPassword: false,
+      emailError: false,
```

Comment:
Only 1 state out of `emailError` and `emailErrorMessage` would be sufficient. Use `emailErrorMessage === ''` for falsy values. Do the same for other state variables as well